pkg/spec/deployment.go

```diff
@@ -56,6 +56,9 @@ func (deployment *DeploymentSpecMod) Fix() error {
 
 	deployment.ControllerFields.ObjectMeta.Labels = addKeyValueToMap(appLabelKey, deployment.ControllerFields.Name, deployment.ControllerFields.ObjectMeta.Labels)
 
+	if deployment.ControllerFields.Appversion != "" {
+		deployment.ControllerFields.ObjectMeta.Annotations = addKeyValueToMap(appVersion, deployment.ControllerFields.Appversion, deployment.ControllerFields.ObjectMeta.Annotations)
+	}
```

Comment:
If you do https://github.com/kedgeproject/kedge/pull/454#discussion_r152549094 this is not needed!